mod/photos.php

```diff
@@ -1332,25 +1332,25 @@ function photos_content(App $a)
 			$result = Item::selectForUser($link_item['uid'], Item::ITEM_FIELDLIST, $condition, $params);
 			$items = Item::inArray($result);
 
-			if (local_user() && (local_user() == $link_item['uid'])) {
+			if (local_user() == $link_item['uid']) {
 				Item::update(['unseen' => false], ['parent' => $link_item['parent']]);
 			}
+		}
 
-			if ($link_item['coord']) {
-				$map = Map::byCoordinates($link_item['coord']);
-			}
+		if (!empty($link_item['coord'])) {
+			$map = Map::byCoordinates($link_item['coord']);
 		}
 
 		$tags = null;
 
-		if (count($linked_items) && strlen($link_item['tag'])) {
```

Comment:
Just a question: was the count on a non existing variable the rootcause of this bug? Because there is no other occurrence of `$linked_items` in this PR.

Reply:
No, the problem came from a discrepancy between `$linked_items` that was populated while `$link_item` was not. The previous code assumed both had to be populated at the same time, the new code checks for the specific `$link_item` fields that are used in each condition block.